recordset/recordset.js

```diff
@@ -317,17 +339,24 @@
         }
 
         ERMrest.resolve(ermrestUri, {cid: context.appName}).then(function getReference(reference) {
-            $rootScope.reference = reference; // TODO contextualize for recordset
+            $rootScope.reference = reference.contextualize.compact;
             $log.info("Reference:", $rootScope.reference);
 
-            recordsetModel.tableDisplayName = reference.displayname;
-            recordsetModel.columns = reference.columns;
+            recordsetModel.tableDisplayName = $rootScope.reference.displayname;
+            recordsetModel.columns = $rootScope.reference.columns;
 
             return $rootScope.reference.read(pageInfo.pageLimit);
         }).then(function getPage(page) {
             recordsetModel.page = page;
             recordsetModel.rowValues = page.tuples.map(function(tuple, index, array) {
-                return tuple.values;
+                var row = [];
+                tuple.values.forEach(function(value, index) {
+                    row.push({
+                        isHTML: tuple.isHTML[index],
+                        value: (tuple.isHTML[index]? $sce.trustAsHtml(value) : value)
+                    });
+                });
+                return row;
```

Comment:
This is the same block of code that is duplicated in the controller, maybe there's a way that you could factor it out and re-use it in both places. A common utility might make sense because this is duplicated in record-two also.